lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/evaluator/parameters/LocalThreadMode.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.reef.runtime.common.evaluator.parameters;
+
+import org.apache.reef.tang.annotations.Name;
+import org.apache.reef.tang.annotations.NamedParameter;
+
+/**
+ * Flag used to set whether we are running in local thread mode or not.
+ */
+@NamedParameter(doc = "Flag for local thread mode.", default_value = LocalThreadMode.NO)
+public final class LocalThreadMode implements Name<String> {
```

Comment:
Yes. I found that other parameters (e.g., `ClientPresent`) used this design so I also embraced it. Do you prefer to go with Boolean?